src/compas_timber/connections/joint.py

```diff
@@ -127,3 +127,26 @@ def create(cls, assembly, *beams):
         assembly.add_joint(joint, beams)
         joint.add_features()
         return joint
+
+    @property
+    def ends(self):
+        self._ends = {}
+        for index, beam in enumerate(self.beams):
+            start_distance = min(
+                [
+                    beam.centerline.start.distance_to_point(self.beams[index - 1].centerline.start),
+                    beam.centerline.start.distance_to_point(self.beams[index - 1].centerline.end),
+                ]
+            )
+            end_distance = min(
+                [
+                    beam.centerline.end.distance_to_point(self.beams[index - 1].centerline.start),
+                    beam.centerline.end.distance_to_point(self.beams[index - 1].centerline.end),
+                ]
+            )
+            if start_distance < end_distance:
+                self._ends[str(beam.key)] = "start"
+            else:
+                self._ends[str(beam.key)] = "end"
```

Comment:
I think this can be written a bit more compact. there are always only two beams so you could unpack and then avoid the indexing